tests/app/services/test_request_supplementary_data.py

```diff
@@ -185,3 +223,47 @@ def test_get_supplementary_data_max_retries(app: Flask, mocker):
 
     assert str(exc.value) == "Supplementary Data request failed"
     assert mocked_make_request.call_count == 3
+
+
+def test_decrypt_supplementary_data_decrypts_when_encrypted_payload_is_valid(
+    app: Flask,
+):
+    with app.app_context():
+        result = decrypt_supplementary_data(encrypted_mock_supplementary_data_payload)
+        assert result == decrypted_mock_supplementary_data_payload
+
+
+def test_decrypt_supplementary_data_raises_validation_error_when_encrypted_payload_missing_data(
+    app: Flask,
+):
+    with app.app_context():
+        with pytest.raises(ValidationError) as e:
+            decrypt_supplementary_data(mock_supplementary_data_payload_missing_data)
+        assert EXPECTED_SDS_DECRYPTION_VALIDATION_ERROR in str(e.value)
```

Comment:
super minor, instead of casting, for an exact match you could assert the expected error is one of the items in e.value.messages

Reply:
or equal to e.value.messages[0]

Reply:
Have changed to ` assert EXPECTED_SDS_DECRYPTION_VALIDATION_ERROR in e.value.messages`